pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -2351,63 +2351,44 @@ See the property `annotations`.
             <property name="xpath">
                 <value>
 <![CDATA[
-//ClassOrInterfaceDeclaration
-    [pmd-java:hasAnnotation('lombok.experimental.UtilityClass')]
-    [ClassOrInterfaceBody
-        [
-            (
-                (: no methods :)
-                not(MethodDeclaration)
-                or
-                (: only private methods :)
-                count(MethodDeclaration) = count(MethodDeclaration[@Private=true()])
-            )
-            and
-            (
-                (: no fields :)
-                not(FieldDeclaration)
-                or
-                (: only private fields :)
-                count(FieldDeclaration) = count(FieldDeclaration[@Private=true()])
-
-            )
-       ]
+let $topLevelClass := //ClassOrInterfaceDeclaration[@Nested= false() and @Local= false()] return
+let $isLombokUtility := exists($topLevelClass[pmd-java:hasAnnotation('lombok.experimental.UtilityClass')]) return
+$topLevelClass[
```

Comment:
```suggestion
let $topLevelClass := /*/ClassOrInterfaceDeclaration return
let $isLombokUtility := exists($topLevelClass[pmd-java:hasAnnotation('lombok.experimental.UtilityClass')]) return
$topLevelClass[
```
There may be several top level classes and isLombokUtility tests them all.

Honestly I think this rule is too complex to be an XPath rule and we should rewrite it in Java... I'll merge this and try my hand at it